system/lib/wasmfs/file.cpp

```diff
@@ -23,35 +23,6 @@ std::shared_ptr<File> Directory::Handle::getEntry(std::string pathName) {
   }
 }
 //
-// MemoryFile
-//
-__wasi_errno_t MemoryFile::write(const uint8_t* buf, size_t len, off_t offset) {
-  if (offset + len >= buffer.size()) {
-    buffer.resize(offset + len);
-  }
-  memcpy(&buffer[offset], buf, len);
-
-  return __WASI_ERRNO_SUCCESS;
-}
-
-__wasi_errno_t MemoryFile::read(uint8_t* buf, size_t len, off_t offset) {
-  assert(offset + len - 1 < buffer.size());
-  std::memcpy(buf, &buffer[offset], len);
-
-  return __WASI_ERRNO_SUCCESS;
-}
-
-void MemoryFile::Handle::preloadFromJS(int index) {
```

Comment:
It would be good to move these implementations to a new `memory_file.cpp` rather than moving them into a header.